File: scripts/analyze_pa.mjs
import { readFileSync } from 'fs';
import XLSX from 'xlsx';

const wb = XLSX.readFile('data/imports/pa_wine_catalog.xlsx');
const data = XLSX.utils.sheet_to_json(wb.Sheets[wb.SheetNames[0]]);

// Group breakdown
const groups = {};
data.forEach(r => { const g = r['Group Name'] || 'null'; groups[g] = (groups[g] || 0) + 1; });
console.log('Group Names:', JSON.stringify(groups, null, 2));
console.log('Total rows:', data.length);

// UPC coverage
let hasUpc = 0;
let totalUpcs = 0;
data.forEach(r => {
  const keys = Object.keys(r).filter(k => k === 'UPC' || k.startsWith('UPC'));
  const upcs = keys.map(k => r[k]).filter(Boolean);
  if (upcs.length > 0) { hasUpc++; totalUpcs += upcs.length; }
});
console.log('\nRows with UPC:', hasUpc, '/', data.length);
console.log('Total UPC values:', totalUpcs);

// Vintage coverage
let hasVintage = 0;
data.forEach(r => {
  if (r['Vintage'] && r['Vintage'] !== 'Nonvintage') hasVintage++;
});
console.log('Rows with vintage:', hasVintage);

// Country/region
const countries = {};
data.forEach(r => { const c = r['Country'] || 'null'; countries[c] = (countries[c] || 0) + 1; });
console.log('\nCountries:', JSON.stringify(countries));

// Sample real wine with UPC and vintage
const sample = data.find(r => r['Vintage'] && r['Vintage'] !== 'Nonvintage' && r['Region'] && r['UPC']);
console.log('\nSample wine with UPC + vintage:', JSON.stringify(sample, null, 2));

File: pipeline/analyze/analyze_pa.py
#!/usr/bin/env python3
"""
Analyze PA PLCB wine catalog Excel file.

Usage:
    python -m pipeline.analyze.analyze_pa
"""

import json
import sys
from collections import Counter
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2]))

import openpyxl

DATA_DIR = Path(__file__).resolve().parents[2] / "data" / "imports"


def main():
    wb = openpyxl.load_workbook(DATA_DIR / "pa_wine_catalog.xlsx", read_only=True)
    ws = wb.active
    all_rows = list(ws.iter_rows(values_only=True))
    wb.close()

    headers = all_rows[0]
    data = [dict(zip(headers, r)) for r in all_rows[1:]]

    groups = Counter(r.get("Group Name") or "null" for r in data)
    print("Group Names:", json.dumps(dict(groups), indent=2))
    print(f"Total rows: {len(data)}")

    has_upc = 0
    total_upcs = 0
    for r in data:
        upcs = [r[k] for k in r if k and (k == "UPC" or str(k).startswith("UPC")) and r[k]]
        if upcs:
            has_upc += 1
            total_upcs += len(upcs)
    print(f"\nRows with UPC: {has_upc} / {len(data)}")
    print(f"Total UPC values: {total_upcs}")

    has_vintage = sum(1 for r in data if r.get("Vintage") and r["Vintage"] != "Nonvintage")
    print(f"Rows with vintage: {has_vintage}")

    country_counter = Counter(r.get("Country") or "null" for r in data)
    print(f"\nCountries: {json.dumps(dict(country_counter))}")

    sample = next((r for r in data if r.get("Vintage") and r["Vintage"] != "Nonvintage" and r.get("Region") and r.get("UPC")), None)
    if sample:
        print(f"\nSample wine with UPC + vintage: {json.dumps(sample, indent=2, default=str)}")


if __name__ == "__main__":
    main()
